static/app/views/issueList/editableTabTitle.tsx

```diff
@@ -0,0 +1,120 @@
+import {useCallback, useEffect, useRef, useState} from 'react';
+import styled from '@emotion/styled';
+
+import useKeyPress from 'sentry/utils/useKeyPress';
+import useOnClickOutside from 'sentry/utils/useOnClickOutside';
+
+function EditableTabTitle({
+  label,
+  onChange,
+  isEditing,
+  setIsEditing,
+  inputRef,
+}: {
+  isEditing: boolean;
+  label: string;
+  onChange: (newLabel: string) => void;
+  setIsEditing: (isEditing: boolean) => void;
+  inputRef?: React.RefObject<HTMLInputElement>;
+}) {
+  const [inputValue, setInputValue] = useState(label);
+
+  const wrapperRef = useRef<HTMLDivElement>(null);
+
+  const enter = useKeyPress('Enter', undefined, true);
```

Comment:
Looks like this is how the existing `EditableText` component does this, but it would be better I think to do this in the `onKeyDown` handler on the input